meinberlin/react/plans/use-create-search-profile.js

```diff
@@ -0,0 +1,171 @@
+import { useState } from 'react'
+import django from 'django'
+import { updateItem } from '../contrib/helpers'
+
+const searchProfileText = django.gettext('Search Profile')
+const projectsInText = django.gettext('Projects in')
+const errorCreatingSearchProfile = django.gettext(
+  'Error creating search profile'
+)
+
+const STATUS_MAPPING = {
+  running: 'active',
+  done: 'past',
+  future: 'future'
+}
+
+export function useCreateSearchProfile ({
+  searchProfilesApiUrl,
+  appliedFilters,
+  districts,
+  organisations,
+  topicChoices,
+  participationChoices,
+  projectStatus,
+  onSearchProfileCreate
+}) {
+  const [loading, setLoading] = useState(false)
+  const [error, setError] = useState(null)
+
+  const createSearchProfile = async () => {
+    setLoading(true)
+    setError(null)
+
+    const results = getFilteredResults({
+      appliedFilters,
+      districts,
+      organisations,
+      topicChoices,
+      participationChoices,
+      projectStatus
+    })
+
+    const name = generateName({ ...results, search: appliedFilters.search })
+
+    const {
+      districtIds,
+      organisationIds,
+      topicIds,
+      participationIds,
+      projectStatusIds
+    } = getFilteredIds(results)
+
+    const params = {
+      name,
+      districts: districtIds,
+      organisations: organisationIds,
+      topics: topicIds,
+      project_types: participationIds,
+      status: projectStatusIds
+    }
+
+    if (appliedFilters.search.length > 0) {
+      params.query_text = appliedFilters.search
+    }
+
+    try {
+      const response = await updateItem(params, searchProfilesApiUrl, 'POST')
+      const searchProfile = await response.json()
+
+      onSearchProfileCreate(searchProfile)
+    } catch (err) {
+      setError(errorCreatingSearchProfile)
+    } finally {
+      setLoading(false)
+    }
+  }
+
+  return { loading, error, createSearchProfile }
+}
+
+function getFilteredResults ({
+  appliedFilters,
+  districts,
+  organisations,
+  topicChoices,
+  participationChoices,
+  projectStatus
+}) {
+  const filters = [
+    districts.filter(district =>
+      appliedFilters.districts.includes(district.name)
+    ),
+    organisations.filter(organisation =>
+      appliedFilters.organisation.includes(organisation.name)
+    ),
+    topicChoices.filter(topic =>
+      appliedFilters.topics.includes(topic.code)
+    ),
+    appliedFilters.participations.map(index =>
+      participationChoices[index]
+    ),
+    projectStatus.filter(status =>
+      appliedFilters.projectState.includes(STATUS_MAPPING[status.name])
+    )
+  ]
+
+  const [
+    filteredDistricts,
+    filteredOrganisations,
+    filteredTopics,
+    filteredParticipationChoices,
+    filteredProjectStatus
+  ] = filters
+
+  return {
+    filteredDistricts,
+    filteredOrganisations,
+    filteredTopics,
+    filteredParticipationChoices,
+    filteredProjectStatus
+  }
```

Comment:
```suggestion
  return {
    filteredDistricts: districts.filter(district =>
      appliedFilters.districts.includes(district.name)
    ),
    filteredOrganisations:  organisations.filter(organisation =>
      appliedFilters.organisation.includes(organisation.name)
    ),
    filteredTopics: topicChoices.filter(topic =>
      appliedFilters.topics.includes(topic.code)
    ),
    filteredParticipationChoices: appliedFilters.participations.map(index =>
      participationChoices[index]
    ),
    filteredProjectStatus: projectStatus.filter(status =>
      appliedFilters.projectState.includes(STATUS_MAPPING[status.name])
    )
  }
```

Or is it not the same?